From 85d9c65db3462959a45513d2297a4a6f83efa77b Mon Sep 17 00:00:00 2001
From: "谢致邦 (XIE Zhibang)" <Yeking@Red54.com>
Date: Thu, 27 Aug 2026 12:49:46 +0800
Subject: [PATCH] Upgrade libbrotli to v1.2.0+e230f47

This fixes build issues of some targets.

Ref: https://github.com/google/brotli/commit/e230f474b87134e8c6c85b630084c612057f253e
---
 libbrotli | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Submodule libbrotli 028fb5a23661..e230f474b871:
diff --git a/libbrotli/c/common/platform.h b/libbrotli/c/common/platform.h
index b59f9b8091c2..e1254d5ca2d5 100644
--- a/libbrotli/c/common/platform.h
+++ b/libbrotli/c/common/platform.h
@@ -213,6 +213,10 @@ OR:
 #define BROTLI_TARGET_MIPS64
 #endif
 
+#if defined(__ia64__) || defined(_M_IA64)
+#define BROTLI_TARGET_IA64
+#endif
+
 #if defined(BROTLI_TARGET_X64) || defined(BROTLI_TARGET_ARMV8_64) || \
     defined(BROTLI_TARGET_POWERPC64) || defined(BROTLI_TARGET_RISCV64) || \
     defined(BROTLI_TARGET_LOONGARCH64) || defined(BROTLI_TARGET_MIPS64)
@@ -665,13 +669,14 @@ BROTLI_UNUSED_FUNCTION void BrotliSuppressUnusedFunctions(void) {
 #undef BROTLI_TEST
 #endif
 
-#if BROTLI_GNUC_HAS_ATTRIBUTE(model, 3, 0, 3)
+#if !defined(BROTLI_MODEL) && BROTLI_GNUC_HAS_ATTRIBUTE(model, 3, 0, 3) && \
+    !defined(BROTLI_TARGET_IA64) && !defined(BROTLI_TARGET_LOONGARCH64)
 #define BROTLI_MODEL(M) __attribute__((model(M)))
 #else
 #define BROTLI_MODEL(M) /* M */
 #endif
 
-#if BROTLI_GNUC_HAS_ATTRIBUTE(cold, 4, 3, 0)
+#if !defined(BROTLI_COLD) && BROTLI_GNUC_HAS_ATTRIBUTE(cold, 4, 3, 0)
 #define BROTLI_COLD __attribute__((cold))
 #else
 #define BROTLI_COLD /* cold */
-- 
2.55.0

